trace-mini-agent/src/config.rs

```diff
@@ -14,11 +17,19 @@ pub struct Config {
     pub stats_flush_interval: u64,
     /// timeout for environment verification, in milliseconds
     pub verify_env_timeout: u64,
+    pub trace_intake_url: String,
+    pub trace_stats_intake_url: String,
+    pub dd_site: String,
 }
 
 impl Config {
     pub fn new() -> Result<Config, Box<dyn std::error::Error>> {
-        let api_key = env::var("DD_API_KEY")?;
+        let api_key = match env::var("DD_API_KEY") {
```

Comment:
you could simplify this block I think as you're not doing anything special with res

Reply:
I didn't do the following because I wanted to return a more descriptive error in case the api key doesn't exist:
```
let api_key = env::var("DD_API_KEY")?
```

And the alternative I found was to do the following:
```
let api_key = env::var("DD_API_KEY").map_err(|_| -> Box<dyn std::error::Error> {
    anyhow::anyhow!("DD_API_KEY environment variable is not set").into()
})?;
```

which I thought was less clear than the current solution. was there an alternative you were thinking of?

Reply:
what do you think about:
`let api_key = env::var("DD_API_KEY").map_err(|_| anyhow::anyhow!("DD_API_KEY environment variable is not set"))?;`

Reply:
ooh I like it, I'll do that. Thanks!